src/rpc/handlers/LedgerEntry.cpp

```diff
@@ -157,21 +157,37 @@ LedgerEntryHandler::process(LedgerEntryHandler::Input input, Context const& ctx)
         return Error{*status};
 
     auto const lgrInfo = std::get<ripple::LedgerHeader>(lgrInfoOrStatus);
-    auto const ledgerObject = sharedPtrBackend_->fetchLedgerObject(key, lgrInfo.seq, ctx.yield);
-
-    if (!ledgerObject || ledgerObject->empty())
-        return Error{Status{"entryNotFound"}};
+    std::optional<data::Blob> ledgerObject;
+    auto output = LedgerEntryHandler::Output{};
+    output.deleted_ledger_index = std::nullopt;
+    if (input.include_deleted.has_value() && input.include_deleted.value() == true){
+        auto const lastTwoObjects = sharedPtrBackend_->fetchLastTwoLedgerObjects(key, lgrInfo.seq, ctx.yield);
+        if (lastTwoObjects.empty()) {
+            return Error{Status{"entryNotFound"}};
+        }
+        else if (lastTwoObjects.size() == 1) {
+            ledgerObject = std::make_optional(lastTwoObjects[0].second);
+            if (!ledgerObject || ledgerObject->empty())
+                return Error{Status{"entryNotFound"}};
+        } 
+        else if (lastTwoObjects.size() == 2) {
+            ledgerObject = lastTwoObjects[0].second.empty()? std::make_optional(lastTwoObjects[1].second) : std::make_optional(lastTwoObjects[0].second);
+            output.deleted_ledger_index = lastTwoObjects[0].second.empty()? lastTwoObjects[0].first : 0;
+        } 
```

Comment:
it might be desirable to add an `else` block to log an error if the number of entries is more than 2, even though it's not possible, but might be still good to check:
`LOG(log_.error()) << "Entries returned exceeded the expected";`